AutomatedTesting/Gem/PythonTests/Prefab/tests/detach_prefab/DetachPrefab_WithNestedEntities.py

```diff
@@ -0,0 +1,76 @@
+        from editor_python_test_tools.prefab_utils import Prefab
+        from editor_python_test_tools.wait_utils import PrefabWaiter
+        import Prefab.tests.PrefabTestUtils as prefab_test_utils
+
+        TEST_PREFAB_FILE_NAME = Path(__file__).stem + '_prefab1'
+        NESTED_ENTITIES_NAME_PREFIX = 'Entity_'
+        NUM_NESTED_ENTITIES_LEVELS = 10
+        POSITION = math.Vector3(0.0, 0.0, 0.0)
+
+        prefab_test_utils.open_base_tests_level()
+
+        # Creates new nested entities at the root level. Asserts if creation didn't succeed
+        nested_entities_root = prefab_test_utils.create_linear_nested_entities(NESTED_ENTITIES_NAME_PREFIX,
+                                                                               NUM_NESTED_ENTITIES_LEVELS, POSITION)
+        level_entity = nested_entities_root.get_parent_id()
+        prefab_test_utils.validate_linear_nested_entities(nested_entities_root, NUM_NESTED_ENTITIES_LEVELS,
+                                                          POSITION)
+
+        # Creates a prefab from the nested entities. Asserts if prefab creation doesn't succeed
+        _, nested_entities_prefab_instance = Prefab.create_prefab([nested_entities_root],
+                                                                  TEST_PREFAB_FILE_NAME)
+        nested_entities_root_on_instance = nested_entities_prefab_instance.get_direct_child_entities()[0]
+        prefab_test_utils.validate_linear_nested_entities(nested_entities_root_on_instance, NUM_NESTED_ENTITIES_LEVELS,
+                                                          POSITION)
+
+        # Detaches the nested entities prefab instance
+        Prefab.detach_prefab(nested_entities_prefab_instance)
+
+        # Test undo/redo on prefab detach
+        general.undo()
+        PrefabWaiter.wait_for_propagation()
+        is_prefab = editor.EditorComponentAPIBus(bus.Broadcast, "HasComponentOfType",
+                                                 nested_entities_prefab_instance.container_entity.id,
+                                                 azlmbr.globals.property.EditorPrefabComponentTypeId)
+        assert is_prefab, "Undo operation failed. Entity hierarchy is not recognized as a prefab."
+        prefab_test_utils.validate_linear_nested_entities(nested_entities_root_on_instance, NUM_NESTED_ENTITIES_LEVELS,
+                                                          POSITION)
+
+        general.redo()
+        PrefabWaiter.wait_for_propagation()
+        former_container_entity = EditorEntity.find_editor_entity(TEST_PREFAB_FILE_NAME)
+        root_entity = EditorEntity.find_editor_entity("Entity_0")
+        is_prefab = editor.EditorComponentAPIBus(bus.Broadcast, "HasComponentOfType", former_container_entity.id,
+                                                 azlmbr.globals.property.EditorPrefabComponentTypeId)
+        assert root_entity.get_parent_id() == former_container_entity.id, \
```

Comment:
Instead of doing this, can we not call validate_linear_nested_entities() too here to validate the hierarchy is same as before?

Reply:
Done!